### DISCUSSION TITLE
Wcf dataportal with asp.net mvc as client

### DISCUSSION THREAD

**Discussion by mohammedsweif** [MarimerLLC/csla#2865] [Questions]
i have wcf datapotal with this configuration

     <service name="Csla.Server.Hosts.WcfPortal" behaviorConfiguration="returnFaults">
        <endpoint binding="customBinding" bindingConfiguration="BinaryCompressionBinding" contract="Csla.Server.Hosts.IWcfPortal" />
      </service>
    </services>
    <bindings>
    <customBinding>
        <binding name="BinaryCompressionBinding">
          <binaryMessageEncoding compressionFormat="GZip">
            <readerQuotas maxBytesPerRead="2147483647" maxArrayLength="2147483647" maxStringContentLength="2147483647" maxNameTableCharCount="2147483647" maxDepth="2147483647" />
          </binaryMessageEncoding>
          <httpTransport maxReceivedMessageSize="2147483647" maxBufferPoolSize="2147483647" maxBufferSize="2147483647" />
        </binding>
      </customBinding>
    </bindings>
    <behaviors>
      <serviceBehaviors>
        <behavior name="returnFaults">
          <serviceDebug includeExceptionDetailInFaults="true" />
          <serviceMetadata httpGetEnabled="true" />
        </behavior>
      </serviceBehaviors>
    </behaviors>
  </system.serviceModel>

---------------------------------
now I need to call this service from asp .net mvc as a client 
how can I configure it
when I configure it as csla dataportal book like  
  add key="CslaDataPortalProxy" value="Csla.DataPortalClient.WcfProxy, Csla" 
     add key="DalManagerType" value="myassemply.DalManager, myassemply.DataLayer"  
    
   add key="csladataportalurl" value="http://localhost:1934/wcfportal.svc" 

The remote server returned an error: (415) Cannot process the message because the content type 'text/xml; charset=utf-8' was not the expected type 'application/soap+msbin1+gzip'..

![image](https://user-images.githubusercontent.com/58108087/161394624-01ab0c93-e501-4d44-8dcb-e3b62bf1e42d.png)

**Comment by rockfordlhotka** [comment]
WCF has no future in .NET. Microsoft didn't implement it in .NET Core (.NET 5, .NET 6, etc.). As a result, your long-term strategy should be to migrate off WCF to another network transport.

The primary data portal channel in CSLA today is `HttpProxy`, where the server is usually hosted in an ASP.NET or aspnetcore web site as an http endpoint. That endpoint is most easily implemented as a controller, and CSLA provides a pre-built data portal controller to make it easy to set up that endpoint.

If you can't switch to http, you'll need to continue trying to debug the WCF issue you are describing.

I would recommend using fiddler or some similar tool to watch the traffic on the network. It sounds like the server is returning an ASCI error message, and the data portal is attempting to treat it like a binary stream (which is what it expects).

Usually this sort of thing happens because the server is rejecting the call due to a security issue or something like that.

**Comment by mohammedsweif** [comment]
Thanks تم الإرسال من هاتف Huawei الخاص بي-------- الرسالة الأصلية --------من:‏ Rockford Lhotka ***@***.***>التاريخ:‏ الاثنين، ٤ أبريل، ٢٠٢٢ ٧:٠٨ مإلى:‏ MarimerLLC/csla ***@***.***>نسخة إلى:‏ mohammedsweif ***@***.***>, Manual ***@***.***>الموضوع:‏ Re: [MarimerLLC/csla] Wcf dataportal with asp.net mvc as client (Discussion #2865)
WCF has no future in .NET. Microsoft didn't implement it in .NET Core (.NET 5, .NET 6, etc.). As a result, your long-term strategy should be to migrate off WCF to another network transport.
The primary data portal channel in CSLA today is HttpProxy, where the server is usually hosted in an ASP.NET or aspnetcore web site as an http endpoint. That endpoint is most easily implemented as a controller, and CSLA provides a pre-built data portal controller to make it easy to set up that endpoint.
If you can't switch to http, you'll need to continue trying to debug the WCF issue you are describing.
I would recommend using fiddler or some similar tool to watch the traffic on the network. It sounds like the server is returning an ASCI error message, and the data portal is attempting to treat it like a binary stream (which is what it expects).
Usually this sort of thing happens because the server is rejecting the call due to a security issue or something like that.
